### ISSUE TITLE
AnalyticalTable: Column Filter indicator is not seen

### ISSUE THREAD

**Issue by PriyaPowar** [SAP/ui5-webcomponents-react#4122]
Hi Colleagues,

We are using AnalyticalTable control with one of the column having a custom filter. We have a usecase that whenever user filters the dat by table column, then the filtered value is stored in a local state. However, on doing so, the filter 'funnel' icon is not seen in on the table column header.

Steps to reproduce the issue:
1) Open sandbox https://codesandbox.io/s/dynamicpage-flexiblecolumn-forked-xxsuyq?file=/src/App.js
2) Click on 'Column1' Header and from the dropdown custom filter , select any value.
3) No filter icon is seen on the table icon but the state is set. Also, in the filter dropdown inputbox, the selected value is also seen.

Please help us in solving this issue.

Regards,
Priya

**Comment by Lukas742** [comment]
Hi @PriyaPowar 

the internal state is reset, as the columns are changing with each filter change. You can use the `reactTableOption` `autoResetFilters` to control that. You can find more information about this in our [documentation](https://sap.github.io/ui5-webcomponents-react/?path=/docs/data-display-analyticaltable--default-story#how-to-stop-the-table-state-from-automatically-resetting-when-the-data-changes).

--> [codeSandbox](https://codesandbox.io/s/dynamicpage-flexiblecolumn-forked-do1cx9?file=/src/App.js:726-780)

**Comment by PriyaPowar** [comment]
Hi @Lukas742 , 
By setting the 'autoResetFilters' to false, I noticed that the filter icon appears. However, it also filters the data in the table again. In our case, this is causing an issue. 

our main dataset is as follows:
[ {docNo: '1001', status: 1 },
  {docNo: '1002', status: 2 },
  {docNo: '1003', status: 1 } ]

status value help dataset:
[ { status: 1, statusText: 'released'},
  { status: 2, statusText: 'unreleased'} ]

the table columns are:
[ {accessor: docNo},
  {accessor:status,
   cell: props => (
          <>
          <Text>
          {valueHelpDataset.find(v=>v.status ===  props.row.original.status).statusText}
          </Text>
          </>)
  }
]

The column on which the filter is present is an ID column(eg: status as the accessor property), however, we display the corresponding text (eg:statusText obtained by key-value mapping) in it. We also show the text (eg: statusText) in custom filter dropdown options. But when user selects any option then, we map the text to its id and fire an api call passing the id as a filter. Now, once the filtered dataset is brought in, I noticed that the table filters the data again but tries to find the text data in id dataset and we get empty rows.

This error could be reproduced for any column wherein the accessor points to one field in dataset and we populate the customfilter value property with its corresponding text value. It tries to search the text data in accessor field of rows.original array.
